### PR TITLE
Correctly handle inline channels in messages 

### LINKED ISSUE #15: Inline channel parser easily breaks
A message like `&lt;<span` will easily cause it to fail. `<[color code here]test>` will too.

### PR DESCRIPTION
**Pull request by xPaw**
Moves inline channel parser into Handlebars, which means it is parsed before parsing colours and links, should fix a couple of parser related bugs.

### CODE REVIEW

**Comment by astorije** [comment]
Sadly, I am :-1: on this. One of the networks I'm on almost exclusively uses `&channels` (it's not a personal/small IRC network btw).
Are there any other alternatives we could explore?

**Comment by xPaw** [comment]
What network does that?

**Comment by astorije** [comment]
> What network does that?

irc.w3.org :P

**Comment by tguild** [comment]
W3C is following this project and strongly contemplating replacing our qwebirc instance

**Comment by astorije** [comment]
> I might have a better fix for this in mind, will try.

If you do, might want to consider closing and opening a new PR instead of ammending commit to keep change history :-)

**Comment by xPaw** [comment]
Okay, pushed a proper fix.

**Comment by dgw** [comment]
Wouldn't it be better to parse the `CHANTYPES` sent by the server? Maybe not in this PR, but eventually.

**Comment by xPaw** [comment]
> Wouldn't it be better to parse the CHANTYPES sent by the server? Maybe not in this PR, but eventually.

Yes, that's not a terrible idea. #24

File: client/js/libs/handlebars/parse.js [hunk 1]
```diff
@@ -48,6 +38,12 @@ function uri(text) {
 	});
 }
 
+function channels(text) {
+	return text.replace(
```
Comment (astorije) [inline client/js/libs/handlebars/parse.js]: Could you re-add here the comment you deleted below please?

```js
// Channels names are strings (beginning with a '&' or '#' character)
// of length up to 200 characters.
// See https://tools.ietf.org/html/rfc1459#section-1.3
```

**Comment by astorije** [comment]
@xPaw, looks nice at first glance, I'll give it a deeper review and an actual test.

Could you comment your code a bit, to be nice with future ourselves, please?
If you also add some details to the commit message on top of the comment, well that would be truly terrific :-)

**Comment by xPaw** [comment]
I've pushed a fix to correctly handle characters that cannot be in the channel name before the `#` (like commas).

**Comment by maxpoulin64** [comment]
Code looks good to me and it works very well. Also fixes a whole world of broken link parsing so I like it. :+1:

**Comment by astorije** [comment]
:+1:

**Pull request merged**